spring/src/main/java/io/kubernetes/client/spring/extended/controller/KubernetesReconcilerProcessor.java

```diff
@@ -85,13 +85,22 @@ private Controller buildController(SharedInformerFactory sharedInformerFactory,
         Function<?, Request> workQueueKeyFunc = watch.workQueueKeyFunc().newInstance();
         for (Method method : r.getClass().getMethods()) {
           if (method.isAnnotationPresent(AddWatchEventFilter.class)) {
-            addFilter = new AddFilterAdaptor(r, method);
+            AddWatchEventFilter annotation = method.getAnnotation(AddWatchEventFilter.class);
+            if (watch.apiTypeClass().equals(annotation.apiTypeClass())) {
+              addFilter = new AddFilterAdaptor(r, method);
+            }
```

Comment:
This will silently drop non `apiType` types, right? I feel like that might confuse users, perhaps throw an exception? Or log an error?

Reply:
added warning logging if any dangling filter annotation were found